cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/join/Join.java

```diff
@@ -98,6 +96,10 @@ private void getCartesianProduct(List<List<JoinElement<INPUT_RECORD>>> list, int
       return;
     }
 
+    if (index >= list.size()) {
```

Comment:
While testing outer join, I was getting ArrayIndexOutOfBoundsException for the case where list does not have any record from requiredInputs. So the condition for [emit](https://github.com/caskdata/cdap/blob/711be796eaf09346cbcfe2c793480b1d85751437/cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/join/Join.java#L94) was not getting satisfied even at the end of the list. This condition will prevent that.